src/ApiService/ApiService/onefuzzlib/Auth.cs

```diff
@@ -1,45 +1,65 @@
 ﻿namespace Microsoft.OneFuzz.Service;
-
-using System.Buffers.Binary;
 using System.Diagnostics;
-using System.Security.Cryptography;
+using System.IO;
 
 public class Auth {
 
-    private static ReadOnlySpan<byte> SSHRSABytes => new byte[] { (byte)'s', (byte)'s', (byte)'h', (byte)'-', (byte)'r', (byte)'s', (byte)'a' };
+    private static ProcessStartInfo SshKeyGenProcConfig(string tempFile) {
 
-    private static byte[] BuildPublicKey(RSA rsa) {
-        static Span<byte> WriteLengthPrefixedBytes(ReadOnlySpan<byte> src, Span<byte> dest) {
-            BinaryPrimitives.WriteInt32BigEndian(dest, src.Length);
-            dest = dest[sizeof(int)..];
-            src.CopyTo(dest);
-            return dest[src.Length..];
+        string keyGen = "ssh-keygen";
+        var winAzureFunctionPath = "C:\\Program Files\\Git\\usr\\bin\\ssh-keygen.exe";
+        if (File.Exists(winAzureFunctionPath)) {
+            keyGen = winAzureFunctionPath;
         }
+        var p = new ProcessStartInfo() {
+            FileName = keyGen,
+            CreateNoWindow = false,
+            UseShellExecute = false,
+            RedirectStandardOutput = true,
+            RedirectStandardError = true
+        };
+        p.ArgumentList.Add("-t");
+        p.ArgumentList.Add("rsa");
+        p.ArgumentList.Add("-f");
+        p.ArgumentList.Add(tempFile);
+        p.ArgumentList.Add("-P");
+        p.ArgumentList.Add("");
+        p.ArgumentList.Add("-b");
+        p.ArgumentList.Add("2048");
```

Comment:
You can initialize these in the object like so:

```suggestion
            RedirectStandardError = true,
            ArgumentList = {
                "-t",
                "rsa",
                "-f",
                tempFile,
                "-P",
                "",
                "-b",
                "2048"
            },
        };
```

Reply:
I did this initially and it did not work since ArgumentList does not have "public set"